core/finality-grandpa/src/until_imported.rs

```diff
@@ -165,28 +182,35 @@ impl<Block: BlockT, Status, I, M> Stream for UntilImported<Block, Status, I, M>
 
 		if update_interval {
 			let mut known_keys = Vec::new();
-			for (&block_hash, &mut (ref mut last_log, ref v)) in &mut self.pending {
+			for (&block_hash, &mut (block_number, ref mut last_log, ref v)) in &mut self.pending {
 				if let Some(number) = self.status_check.block_number(block_hash)? {
+					// TODO: Now that we pass the number all the way down here, should we check if it is the same as we
+					// are expecting it to be?
```

Comment:
This will be checked later on in `wait_completed`, we pass the canon number to it (the one we fetch here and store in `known_keys`. The `wait_completed` can look at the vote itself and the target number to figure out what the original number was.